Feature/Task-18 Integration between UI and Web APIs.

2musime opened the pull request:
# PR: Integration between UI and Web APIs.
## Description
This Pull Request is designed to integrate the UI project (consumer) to the Web APIs so it works seamlessly as a whole ..**..Certificate-Manager-App..**..

## Changes:

- [x] Implemented API calls for fetching and searching users, suppliers, and certificates using their respective endpoints in the controllers.

2musime committed:
Feature/Task-18 Users and certificate can be loaded in UI

2musime committed:
Feature/Task-18 Users and certificate can be loaded in UI

2musime committed:
Feature/Task-18 Delete request is working

2musime committed:
Feature/Task-18 Delete request is working

2musime committed:
Feature/Task-18: Supplier functionality is seamless integrated.

2musime committed:
Feature/Task-18: Supplier functionality is seamless integrated.

2musime committed:
Feature/Task-18 Participants can be loaded

2musime committed:
Feature/Task-18 Participants can be loaded

2musime committed:
Feature/Task-18 made changes on the database structure to accept pdf file
     - Made change on CertificateDto

2musime committed:
Feature/Task-18 made changes on the database structure to accept pdf file

2musime committed:
Feature/Task-18 made changes on the database structure to accept pdf file
     - Made change on CertificateDto

2musime committed:
Merge branch 'Feature/Task-18-integration-between-ui-and-apis' of https://github.com/Tumusime14/Certificate-Manager---David into Feature/Task-18-integration-between-ui-and-apis

2musime committed:
Feature/Task-18: Enhancing certificate method

2musime committed:
Feature/Task-18: Enhancing certificate method

2musime committed:
Feature/Task-18: Creating and edit is bit working beside some bugs

2musime committed:
Feature/Task-18: Creating and edit is bit working beside some bugs

2musime committed:
- Posting request with participants

2musime committed:
- Created new controller for comment request
- Refactored the certificate POST request load,
- Get request for all certificate details correctly fixed.

2musime committed:
Merge branch 'Feature/Task-18-integration-between-ui-and-apis' of https://github.com/Tumusime14/Certificate-Manager---David into Feature/Task-18-integration-between-ui-and-apis

2musime committed:
Feature/Task18: - Fetch Certificate Data,
- Set State with Fetched Data
- Refactored DTOs

elhasic reviewed (changes requested):
Fetching data from API ca not use like this:
const response = await fetch('https://localhost:7164/api/Certificate');
const data = await response.json();
Please add separate file where you will define inicilatiosation of API

2musime committed:
- Renamed controllers to follow plural naming conventions for consistency across API endpoints.
- Updated CertificateService logic for improved certificate management.
- Refined modal components for comments, participants, and suppliers in the frontend.
- Adjusted UserSwitcher and certificate-related pages to support updated naming and functionality.
